src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -373,3 +373,21 @@ function syzygy_generators(
   return elem_type(F)[F(s[i]) for i=1:Singular.ngens(s)]
 end
 
+function symbols(F::FreeMod)
+  return _get_symbols!(F, F.S)
+end
+
+function symbol(F::FreeMod, i::Int)
+  return symbols(F)[i]
+end
+
+function _get_symbols!(F::FreeMod, S::Vector{Symbol})
+  return S
+end
+
+function _get_symbols!(F::FreeMod, symbol_fun::Function)
+  F.S = symbol_fun()::Vector{Symbol}
+  return F.S
```

Comment:
```suggestion
  F.S = symbol_fun()
  return F.S::Vector{Symbol}
```
to make this type stable